src/app.ts

```diff
@@ -236,11 +237,36 @@ export class App {
    *   ctx.res.set('X-Mojo', 'I <3 mojo.js!');
    *   await ctx.render(...args);
    * });
+   *
+   * // Render response with header using nested helper
+   * app.addHelper('renderWith.header', async (ctx, ...args) => {
+   *   ctx.res.set('X-Mojo', 'I <3 mojo.js!');
+   *   await ctx.render(...args);
+   * });
    */
   addHelper(name: string, fn: MojoAction): this {
-    return this.decorateContext(name, function (this: MojoContext, ...args: any[]) {
-      return fn(this, ...args);
-    });
+    const nestedNames = name.split('.');
+    if (nestedNames.length == 1)
+      return this.decorateContext(name, function (this: MojoContext, ...args: any[]) {
+        return fn(this, ...args);
+      });
+    else if (nestedNames.length == 2) {
+      const [getterName, methodName] = nestedNames;
+      this._nestedFunctions[methodName] = fn;
```

Comment:
This approach seems broken, since it only works for a single getter name. Probably needs a test with at least two nested helpers like `ctx.foo.testHelper()` and `ctx.bar.testHelper()`.

Reply:
fixed in last push